core/src/main/java/tech/pegasys/web3signer/core/multikey/metadata/parser/YamlSignerParser.java

```diff
@@ -50,6 +51,9 @@ public YamlSignerParser(final Collection<AbstractArtifactSignerFactory> signerFa
       final SigningMetadata metaDataInfo =
           OBJECT_MAPPER.readValue(metadataPath.toFile(), SigningMetadata.class);
 
+      // Custom validation of metadata parameters which is not enforced by Jackson
+      SigningMetadataValidator.validate(metaDataInfo);
```

Comment:
maybe move the validation work into the "createSigner"? At least then the function knows what type is expected.